staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -184,8 +199,38 @@ func (c *AvailableConditionController) sync(key string) error {
 			return err
 		}
 	}
-
-	// TODO actually try to hit the discovery endpoint
+	// actually try to hit the discovery endpoint when it isn't local and when we're routing as a service.
+	// TODO figure out if the kube-proxy code handling has similar problems
+	if apiService.Spec.Service != nil && !c.enableAggregatorRouting && c.serviceResolver != nil {
```

Comment:
> Its not clear to me why we care how the request will be routed (!c.enableAggregatorRouting) for purposes of checking if the user-apiserver is responding.

I'm not actually sure how the condition of "service is present but has invalid endpoints" is handled by the separate aggregatorrouting dialer.  I wasn't sure if it was a problem or not and if it is a problem, then the API needs restructuring to make it reasonable to call out like this.